extensions/lifecycle/CircuitBreakerGroup.js

```diff
@@ -0,0 +1,317 @@
+'use-strict';
+/* eslint no-template-curly-in-string: 0 */
+
+const { CircuitBreaker, BreakerState } = require('breakbeat').CircuitBreaker;
+const { startCircuitBreakerMetricsExport } = require('../../lib/CircuitBreaker');
+
+/**
+ * circuit breaker configuration adapted for the bucket processor
+ * @typedef {Object} BucketProcessorCircuitBreakerConfig
+ * @property {Object} global - global circuit breaker configurations (circuit breaks everything)
+ * @property {CircuitBreakerGroup} circuitBreakerGroup - circuit breakers per workflow and location/topic
+ */
+
+class CircuitBreakerGroup {
+    constructor(cbGlobalConf, circuitBreakers) {
+        if (circuitBreakers) {
+            this.circuitBreakers = circuitBreakers;
+            return;
+        }
+
+        this._cbGlobalConf = cbGlobalConf;
+
+        // breaker id used for the metrics
+        this._latestBreakerId = 0;
+
+        this.circuitBreakers = {
+            transition: {
+                topic: {},
+                location: {},
+                global: [],
+            },
+            expiration: {
+                topic: {},
+                location: {},
+                global: [],
+            },
+        };
+    }
+
+    addTransitionCircuitBreaker(circuitBreaker, topic, location) {
```

Comment:
nit: "internal" methods should start with an `_` (i.e. most methods except `addCircuitBreaker`)